src/main/java/org/myteam/server/board/service/BoardReplyRecommendService.java

```diff
@@ -0,0 +1,93 @@
+import org.myteam.server.board.domain.BoardReply;
+import org.myteam.server.board.domain.BoardReplyRecommend;
+import org.myteam.server.board.repository.BoardReplyLockRepository;
+import org.myteam.server.board.repository.BoardReplyRecommendRepository;
+import org.myteam.server.global.exception.ErrorCode;
+import org.myteam.server.global.exception.PlayHiveException;
+import org.myteam.server.member.entity.Member;
+import org.myteam.server.member.service.SecurityReadService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class BoardReplyRecommendService {
+
+    private final BoardReplyReadService boardReplyReadService;
+    private final SecurityReadService securityReadService;
+    private final BoardReplyRecommendReadService boardReplyRecommendReadService;
+
+    private final BoardReplyRecommendRepository boardReplyRecommendRepository;
+    private final BoardReplyLockRepository boardReplyLockRepository;
+
+    @Transactional
+    public void recommendBoardReply(Long boardReplyId) {
+        BoardReply boardReply = boardReplyReadService.findById(boardReplyId);
+        Member member = securityReadService.getMember();
+
+        verifyMemberAlreadyRecommend(boardReply, member);
+
+        recommend(boardReply, member);
+
+        addRecommendCount(boardReply.getId());
+    }
+
+    @Transactional
+    public void deleteRecommendBoardReply(Long boardReplyId) {
+        BoardReply boardReply = boardReplyReadService.findById(boardReplyId);
+        Member member = securityReadService.getMember();
+
+        isAlreadyRecommended(boardReply, member);
+
+        boardReplyRecommendRepository.deleteByBoardReplyIdAndMemberPublicId(boardReply.getId(), member.getPublicId());
+
+        minusRecommendCount(boardReply.getId());
+    }
+
+    /**
+     * 추천이 되어있는 상태인지 검사
+     */
+    private void isAlreadyRecommended(BoardReply boardReply, Member member) {
+        boardReplyRecommendReadService.isAlreadyRecommended(boardReply.getId(), member.getPublicId());
+    }
+
+    /**
+     * 이미 추천한 상태인지 검사
+     */
+    private void verifyMemberAlreadyRecommend(BoardReply boardReply, Member member) {
+        boardReplyRecommendReadService.confirmExistBoardReply(boardReply.getId(), member.getPublicId());
+    }
+
+    /**
+     * 게시판 대댓글 추천 생성
+     */
+    private void recommend(BoardReply boardReply, Member member) {
+        BoardReplyRecommend recommend = BoardReplyRecommend.builder()
+                .boardReply(boardReply)
+                .member(member)
+                .build();
+        boardReplyRecommendRepository.save(recommend);
+    }
+
+    /**
+     * 게시판 대댓글 추천수 증가
+     */
+    private void addRecommendCount(Long boardReplyId) {
+        BoardReply reply = boardReplyLockRepository.findById(boardReplyId)
+                .orElseThrow(() -> new PlayHiveException(ErrorCode.BOARD_REPLY_RECOMMEND_NOT_FOUND));
```

Comment:
```suggestion
        BoardReply reply = boardReplyLockRepository.findById(boardReplyId)
                .orElseThrow(() -> new PlayHiveException(ErrorCode.BOARD_REPLY_RECOMMEND_NOT_FOUND));
```

이 부분 중복코드가 보이네요 ReadService에서 findByIdLock메소드를 만들어서 하나로 관리하면 좋을 것 같아요

Reply:
수정했습니다!!